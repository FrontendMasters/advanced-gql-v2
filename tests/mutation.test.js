const gql = require("graphql-tag");
const createTestServer = require("./helper");

const CREATE_POST = gql`
  mutation {
    createPost(input: { message: "hello" }) {
      message
    }
  }
`;

describe("mutation", () => {
  test("createPost", async () => {
    const { mutate } = createTestServer({
      user: { id: 1 },
      models: {
        Post: {
          createOne() {
            return {
              message: "hello"
            };
          }
        },
        user: { id: 1 }
      }
    });

    const res = await mutate({ query: CREATE_POST });
    expect(res).toMatchSnapshot();
  });
});
