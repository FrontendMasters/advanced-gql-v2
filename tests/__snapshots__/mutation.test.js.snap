// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutation createPost 1`] = `
Object {
  "data": Object {
    "createPost": Object {
      "message": "hello",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
